### ISSUE TITLE
Change "seed" to "random_state" in sklearn-like API

### ISSUE THREAD

**Issue by hmartiniano** [dmlc/xgboost#1618]
The "seed" parameter of the XGBClassifier and XGBRegressor classes of the python package should be changed to "random_state" to better follow the sklearn API, as this is the name used for setting the random seed in sklearn's models.
This would improve compatibility with third party packages who expect a sklearn model and try to set the value for a "random_state" parameter, which fails with the xgboost classes.

**Comment by founderfan** [comment]
Also, "nthread" parameter in xgboost sklearn-like api should be changed into "n_jobs", which match sklearn in 0.18

**Comment by terrytangyuan** [comment]
Good call. Feel free to submit PRs for this!

**Comment by gaw89** [comment]
Has anything been done on this?  I would be happy to submit a PR for this.  I am trying to fix the issue right now, but for some reason, when I update sklearn.py and import XGBClassifier, it still give me the parameter "seed" instead of "random_state".  Not sure why...
